docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,134 @@
+users of Kubernetes get surprise, too.
+- Even worse, to implement the same semantics as today's Docker runtime, it will require other runtimes (e.g. rkt) to depend
+on the default capability set as well. And if Docker changes the default capability set, then consistency is not guaranteed.
+- It is arguable that `Add` and `Drop` set are the best semantics for capability set compared to a [whitelist based](https://github.com/appc/spec/blob/af31bda9a474bf3a3e83144f6f0d264cfedc813f/spec/ace.md#oslinuxcapabilities-retain-set) semantics,
+which explicitly sets the bounding capabilities that a container could have, and has less security implications (users don't guess what
+is the final capability set).
+
+## Solutions
+
+As mentioned above, an explicit whitelist type of capability-retain-set could make things very consistent across runtimes and
+version updates.
+But as we can see, it's not very practical for system admins and operators to put a long [default capability list](https://docs.docker.com/engine/reference/run/#/runtime-privilege-and-linux-capabilities) on every
+container he wants to launch.
+Besides, changing the k8s API could take several release cycles.
+
+So in the proposed solution here. We will:
+- Keep the k8s API as is today. Users will not need to change anything.
+- Define a default capability list inside k8s, and document about it.
+- Make CRI expose the whitelist based capability interface, and put the burden on the runtime to implement that.
+- Make the final capability set **visible** through the container status, for example:
+
+```shell
+$ kubectl get pod -o yaml
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: Pod
+  metadata:
+  ...
+  spec:
+    containers:
+    - command:
+      - sleep
+      - "1000"
+      image: busybox
+      imagePullPolicy: Always
+      name: my-busybox
+      securityContext:
+        capabilities:
+          add:
+          - NET_ADMIN
+      ...
+  status:
+    containerStatuses:
+    - image: busybox
+    capabilities:
+      SETPCAP, MKNOD, AUDIT_WRITE, ...
+      name: my-busybox
+      state:
+        running:
+          startedAt: 2016-09-27T22:33:38Z
+    ...
+```
+
+## Open questions:
+
+**Q:** What the default capability list should be?
+
+**A:** According to @smarterclayton, there has been questions and arguments about what the default capability list should be, and it's not settled down yet.
+       So for now, in order to keep backward compatibility, we'd better just define it as [Docker's default capability list](https://docs.docker.com/engine/reference/run/#/runtime-privilege-and-linux-capabilities) today.
+
+**Q:** How wide does the default capability list apply? Does it apply to one node, or the whole cluster?
+
+**A:** I don't really have a use case for support per-node-default default capability list, if someone could give one, it might be very helpful.
+       Other than that, for today's use case, a cluster-wide default capability list is enough.
```

Comment:
no, but I'm also hesitant to say it would be namespaced since it is cluster policy.  What I think could/should be namespaced are the values that drive the PSP.  For example, allocating UIDs/GIDs on a namespace level that drives a dynamic strategy in PSP.

Defining the policy on a per namespace level seems like a big burden for large clusters or we'd have to introduce levels of policy to have defaults if the ns doesn't have policy defined